CodeGeneration/Sources/SyntaxSupport/AttributeNodes.swift

```diff
@@ -267,6 +272,31 @@ public let ATTRIBUTE_NODES: [Node] = [
     ]
   ),
 
+  Node(
+    kind: .abiAttributeArguments,
+    base: .syntax,
+    experimentalFeature: .abiAttribute,
+    nameForDiagnostics: "ABI-providing declaration",
+    documentation: "The arguments of the '@abi' attribute",
+    children: [
+      Child(
+        name: "provider",
+        kind: ChildKind.nodeChoices(choices: [
+          Child(name: "associatedType", kind: .node(kind: .associatedTypeDecl)),
+          Child(name: "declGroup", kind: .node(kind: .declGroupHeader)),
+          Child(name: "deinitializer", kind: .node(kind: .deinitializerDecl)),
+          Child(name: "enumCase", kind: .node(kind: .enumCaseDecl)),
+          Child(name: "function", kind: .node(kind: .functionDecl)),
+          Child(name: "initializer", kind: .node(kind: .initializerDecl)),
+          Child(name: "subscript", kind: .node(kind: .subscriptDecl)),
+          Child(name: "typeAlias", kind: .node(kind: .typeAliasDecl)),
+          Child(name: "variable", kind: .node(kind: .variableDecl)),
+          Child(name: "unsupported", kind: .node(kind: .decl)),
```

Comment:
Instead of having an `unsupported` case here, I think we should put the unsupported declaration into unexpected nodes. That way we can also generate a parser diagnostic for eg. `@api(import Foo)`.

Reply:
Yes, adding a choice for `.missingDecl` sounds like a good solution to me.